system_hooks/src/call_hooks/l1_messenger.rs

```diff
@@ -111,196 +106,60 @@ where
         Err(SystemError::LeafDefect(e)) => Err(e.into()),
     }
 }
-// sendToL1(bytes) - 62f84b24
-pub const SEND_TO_L1_SELECTOR: &[u8] = &[0x62, 0xf8, 0x4b, 0x24];
-
-const L1_MESSAGE_SENT_TOPIC: [u8; 32] = [
-    0x3a, 0x36, 0xe4, 0x72, 0x91, 0xf4, 0x20, 0x1f, 0xaf, 0x13, 0x7f, 0xab, 0x08, 0x1d, 0x92, 0x29,
-    0x5b, 0xce, 0x2d, 0x53, 0xbe, 0x2c, 0x6c, 0xa6, 0x8b, 0xa8, 0x2c, 0x7f, 0xaa, 0x9c, 0xe2, 0x41,
-];
 
 fn l1_messenger_hook_inner<S: EthereumLikeTypes>(
     calldata: &[u8],
     resources: &mut S::Resources,
     system: &mut System<S>,
-    caller: B160,
     _caller_ee: u8,
     is_static: bool,
-) -> Result<Result<Bytes32, &'static str>, SystemError>
+) -> Result<Result<(), &'static str>, SystemError>
 where
 {
     evm_interpreter::charge_native_and_ergs::<S::Resources>(
         resources,
         HOOK_BASE_NATIVE_COST,
-        HOOK_BASE_ERGS_COST,
+        Ergs(0), // Do not charge EVM gas here, it is already charged in L1Messenger smart contract
     )?;
 
-    if calldata.len() < 4 {
+    // Should never happen
+    if is_static {
         return Ok(Err(
-            "L1 messenger failure: calldata shorter than selector length",
+            "L1 messenger failure: sendToL1 called with static context",
         ));
     }
-    let mut selector = [0u8; 4];
-    selector.copy_from_slice(&calldata[..4]);
-    let _ = system
-        .get_logger()
-        .write_fmt(format_args!("Selector for l1 messenger:"));
-    let _ = system.get_logger().log_data(selector.iter().copied());
 
-    match selector {
-        s if s == SEND_TO_L1_SELECTOR => {
-            if is_static {
-                return Ok(Err(
-                    "L1 messenger failure: sendToL1 called with static context",
-                ));
-            }
-
-            send_to_l1_inner(&calldata[4..], resources, system, caller)
-        }
-        _ => Ok(Err("L1 messenger: unknown selector")),
-    }
+    send_to_l1_inner(&calldata, resources, system)
 }
 
-/// Sends a message to L1 and emits the needed events.
-/// Note, that the ABI-encoded event should consist of the following:
-/// 32 bytes offset (must be 32)
-/// 32 bytes length of the message
-/// followed by the message itself, padded to be a multiple of 32 bytes.
+/// Receives calldata in the form of abi.encode(address msg.sender, bytes message)
```

Comment:
We need to update this annotation - it doesn't return anything anymore and ABI is different